### ISSUE TITLE
Method with 6 optional arguments causes fatal error for IDL index

### ISSUE THREAD

**Issue by zcorpan** [tabatkins/bikeshed#572]
As part of fixing https://github.com/whatwg/compat/issues/19 I ran into this bug in bikeshed.

```
interface DOMMatrixReadOnly {
    DOMMatrix scale(optional unrestricted double scaleX = 1,
                    optional unrestricted double scaleY,
                    optional unrestricted double scaleZ = 1,
                    optional unrestricted double originX = 0,
                    optional unrestricted double originY = 0,
                    optional unrestricted double originZ = 0);
};
```
`bikeshed -f spec`
> FATAL ERROR: The argument autolink 'originX' for 'DOMMatrixReadOnly/scale()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'originY' for 'DOMMatrixReadOnly/scale()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.

This doesn't happen if I make the first argument non-optional.

I suppose it's when generating the IDL index, since that's the only thing that autolinks to the argument. But when force-generating it seems to correctly link to it...

**Comment by zcorpan** [comment]
`Boilerplate: omit idl-index` makes that section go away when force-generating, but the fatal errors are still there.

**Comment by zcorpan** [comment]
https://drafts.fxtf.org/bikeshed/geometry/

**Comment by tabatkins** [comment]
How interesting! I'll dig into this in a little bit, but in the meantime: is it "six optional arguments" that cause it, or "all optional arguments"? The latter seems more likely to me.

**Comment by zcorpan** [comment]
Yeah, the latter. More methods in the spec suffer from this now with different number of arguments, but all optional.

**Comment by tabatkins** [comment]
I think I fixed it! Let me know?

**Comment by zcorpan** [comment]
Hmmm nope? For the geometry spec with a newly updated bikeshed:

> $ bikeshed -f spec
FATAL ERROR: The argument autolink 'tx' for 'DOMMatrixReadOnly/translate()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'ty' for 'DOMMatrixReadOnly/translate()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'originX' for 'DOMMatrixReadOnly/scale()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'scale' for 'DOMMatrixReadOnly/scale3d()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'originX' for 'DOMMatrixReadOnly/scale3d()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'originY' for 'DOMMatrixReadOnly/scale3d()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'tx' for 'DOMMatrix/translateSelf()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'ty' for 'DOMMatrix/translateSelf()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'originX' for 'DOMMatrix/scaleSelf()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'scale' for 'DOMMatrix/scale3dSelf()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'originX' for 'DOMMatrix/scale3dSelf()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
FATAL ERROR: The argument autolink 'originY' for 'DOMMatrix/scale3dSelf()' has too many possible overloads to disambiguate. Please specify the full method signature this argument is for.
WARNING: Multiple possible 'constructor' local refs for 'DOMMatrixReadOnly()'.
Arbitrarily chose the one with type 'constructor' and for 'DOMMatrixReadOnly'.
WARNING: Multiple possible 'constructor' local refs for 'DOMMatrix()'.
Arbitrarily chose the one with type 'constructor' and for 'DOMMatrix'.
WARNING: This specification has neither a 'Security Considerations' nor a 'Privacy Considerations' section. Please consider adding both.
